Storage/src/Connection/RetryTrait.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Storage\Connection;
+
+/**
+ * Trait which provides helper methods for retry logic.
+ *
+ * @internal
+ */
+trait RetryTrait
+{
+    /**
+     * The HTTP codes that will be retried by our custom retry function.
+     * @var array
+     */
+    public static $httpRetryCodes = [
+        0, // connetion-refused OR connection-reset gives status code of 0
+        200, // partial download cases
+        408,
+        429,
+        500,
+        502,
+        503,
+        504
+    ];
+
+    /**
+     * The operations which can be retried without any conditions
+     * (Idempotent)
+     * @var array
+     */
+    public static $idempotentOps = [
+        'bucket_acl.get',
+        'bucket_acl.list',
+        'buckets.delete',
+        'buckets.get',
+        'buckets.getIamPolicy',
+        'buckets.insert',
+        'buckets.list',
+        'buckets.lockRetentionPolicy',
+        'buckets.testIamPermissions',
+        'default_object_acl.get',
+        'default_object_acl.list',
+        'hmacKey.delete',
+        'hmacKey.get',
+        'hmacKey.list',
+        'notifications.delete',
+        'notifications.get',
+        'notifications.list',
+        'object_acl.get',
+        'object_acl.list',
+        'objects.get',
+        'objects.list',
+        'serviceaccount.get',
+    ];
+
+    /**
+     * The operations which can be retried with specific conditions
+     * (Conditionally idempotent)
+     * @var array
+     */
+    public static $condIdempotentOps = [
```

Comment:
We should try to make these private. Currently, they are only used in `RetryTrait` and `Connection\Rest` (which implements `RetryTrait`), so they can be private.